### PR TITLE
python3Packages.subprocess-tee: init at 0.3.2

### PR DESCRIPTION
**Pull request by putchar**
###### Motivation for this change
In order to add molecule module in nixpkgs, we need to add subprocess-tee. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8608,6 +8608,12 @@
     githubId = 5636;
     name = "Steve Purcell";
   };
+  putchar = {
+    email = "slim.cadoux@gmail.com";
+    github = "putchar";
+    githubId = 8208767;
+    name = "Slim Cadoux";
+  };
```
Comment (legendofmiracles) [inline maintainers/maintainer-list.nix:8616]: Separate commit: `maintainers: add putchar`

Comment (putchar) [reply maintainers/maintainer-list.nix:8616]: fixed it. thanks

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130576` at d3968ea3 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.subprocess-tee</li>
    <li>python39Packages.subprocess-tee</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-explicit-check-phase.md">python-explicit-check-phase</a>

Consider using `pytestCheckHook` in `checkInputs` rather than `checkPhase = "pytest";`.

Near pkgs/development/python-modules/subprocess-tee/default.nix:27:3:
```
   |
27 |   checkPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/subprocess-tee/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ];
+
+  checkInputs = [
+    pytest
+    enrich
+  ];
+
+  checkPhase = ''
+    pytest
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/subprocess-tee/default.nix:29]:
```suggestion
    pytestCheckHook
    enrich
  ];

```

File: pkgs/development/python-modules/subprocess-tee/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools-scm
+, pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/subprocess-tee/default.nix:5]:
```suggestion
, pytestCheckHook
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/subprocess-tee/default.nix (hunk 2 above)
Comment (putchar) [reply pkgs/development/python-modules/subprocess-tee/default.nix]: thanks. i am doing the change now

File: pkgs/development/python-modules/subprocess-tee/default.nix (hunk 3 above)
Comment (putchar) [reply pkgs/development/python-modules/subprocess-tee/default.nix]: cf above. i am doing the change

File: pkgs/development/python-modules/subprocess-tee/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  checkInputs = [
+    pytestCheckHook
+    enrich
+  ];
+
+  checkPhase = ''
+    pytest
+  '';
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/subprocess-tee/default.nix:30]:
```suggestion
```

Comment (putchar) [reply pkgs/development/python-modules/subprocess-tee/default.nix:30]: i am sorry i though i read your request properly. i am doing the change now.

**Comment by putchar** [comment]
hello people 
is there anything wrong with this PR 
if yes do you think you could find some time hinting me where the issue is ?

Regards

**Review by legendofmiracles — APPROVED** [review]

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by putchar** [comment]
Result of `nixpkgs-review pr 130576` run on x86_64-linux
2 packages built:
python38Packages.subprocess-tee python39Packages.subprocess-tee

**Comment by putchar** [comment]
Result of `nixpkgs-review pr 130576` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.subprocess-tee</li>
    <li>python39Packages.subprocess-tee</li>
  </ul>
</details>

**Review by Kranzes — APPROVED** [review]
LGTM🐒

**Comment by putchar** [comment]
Hello again
Do you have time to review this pr ?
Regards

**Pull request merged**